### PR TITLE
[gcp][feat] OnDemand cost information for GcpDiskType

### PR DESCRIPTION
**Pull request by anjafr**
# Description

OnDemand cost information for GcpDiskType

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/gcp/resoto_plugin_gcp/resources/compute.py [hunk 1]
```diff
@@ -767,6 +767,57 @@ class GcpDiskType(GcpResource, BaseVolumeType):
...
+        log.debug((f"Looking up pricing for {self.rtdname} in {self.region().rtdname}"))
+        resource_group_map = {
+            "local-ssd": "LocalSSD",
+            "pd-balanced": "SSD",
+            "pd-ssd": "SSD",
+            "pd-standard": "PDStandard",
+        }
+        resource_group = resource_group_map.get(self.name)
+
+        def filter(sku: GcpSku) -> bool:
```
Comment (aquamatthias) [inline plugins/gcp/resoto_plugin_gcp/resources/compute.py:784]: filter is a python predefined name. please choose another one.

File: plugins/gcp/resoto_plugin_gcp/resources/compute.py [hunk 2]
```diff
@@ -767,6 +767,58 @@ class GcpDiskType(GcpResource, BaseVolumeType):
     default_disk_size_gb: Optional[str] = field(default=None)
     valid_disk_size: Optional[str] = field(default=None)
 
+    RESOURCE_GROUP_MAP = {
```
Comment (aquamatthias) [inline plugins/gcp/resoto_plugin_gcp/resources/compute.py:770]: This is defined in a data class - it becomes an attribute of every instance if not defined as classvar explicitly.

**Comment by lloesche** [comment]
The current collector has a logic
```
if resource.region(graph).name == "undefined" and resource.zone(graph).name == "undefined":
    log.debug(f"Resource {resource.rtdname} has no region or zone" " - removing from graph")
    graph.remove_node(resource)
    return
```

Is this still implemented somewhere?

**Comment by anjafr** [comment]
> Is this still implemented somewhere?

@lloesche Not in this form, no. We do remove DiskType nodes when they don't have any successors, i.e. aren't used by any actual disk. The new collector defaults to create edges to project if region or zone can't be determined for any resource. I'm not sure on what the initial intention was with the above code. Ultimately we do not keep any unused or otherwise dangling DiskType nodes around.

**Review by aquamatthias — APPROVED** [review]
Thanks. Regarding the question from Lukas: all resources that do not belong to a zone or region are put under the project level. If there are resources that we should not collect, we should do it more explicitly than missing properties.

**Pull request merged**